src/ops/info.rs

```diff
@@ -50,14 +52,20 @@ pub fn info(
 
     validate_locked_and_frozen_options(package_id, config)?;
 
-    let current_rustc_version = &config.load_global_rustc(ws.as_ref())?.version;
+    let msrv_from_nearest_manifest_or_ws =
+        try_get_msrv_from_nearest_manifest_or_ws(config.cwd(), ws.as_ref());
+    // If the workspace does not have a specific Rust version,
+    // or if the command is not called within the workspace, then fallback to the global Rust version.
+    let rustc_version =
+        msrv_from_nearest_manifest_or_ws.unwrap_or(config.load_global_rustc(ws.as_ref())?.version);
     // Remove any pre-release identifiers for easier comparison.
     // Otherwise, the MSRV check will fail if the current Rust version is a nightly or beta version.
-    let current_rustc_version = &semver::Version::new(
-        current_rustc_version.major,
-        current_rustc_version.minor,
-        current_rustc_version.patch,
+    let rustc_version = &semver::Version::new(
+        rustc_version.major,
+        rustc_version.minor,
+        rustc_version.patch,
     );
```

Comment:
Since this is only renevant to `config.load_global_rustc(ws.as_ref())?.version`, we should only do this within that code path

Reply:
Moved! Nice point.